frinx-uniconfig/user-guide/operational-procedures/uniconfig-properties/readme.md

```diff
@@ -0,0 +1,89 @@
+
+The main difference between these two, is that UniConfig Cloud Config Server and 
+Cloud Config Client are in the same project, while Spring requires the use of a separate
+Cloud Config Server application.
+
+It works as follows:
+
+![startup-with-ucc](startup-with-ucc.jpg)
+
+
+* First, before starting UniConfig, we must enable Cloud Config using the following properties
+
+```
+# Uniconfig cloud settings - depends on kafka connection
+uniconfig.cloud.config.enabled=true
+uniconfig.cloud.config.fail-fast=false
+uniconfig.cloud.config.application-name=uniconfig
+spring.cloud.bus.enabled=true
+spring.kafka.bootstrap-servers=http://localhost:9092
+management.endpoints.web.exposure.include=bus-refresh
+# Disable cloud connection to Kafka for testing purpose, property spring.cloud.bus.enabled must be false
+#spring.autoconfigure.exclude=org.springframework.cloud.stream.function.FunctionConfiguration
+```
+
+* When UniConfig starts, it will first check if there are any *default properties* in the database that it could configure
+  * If *default properties* are in the database -> UniConfig instance will use it
+  * If not -> UniConfig instance will use its properties and, once loaded, will save them to a database 
+    for a next UniConfig instances.
+* At the end of Spring initialisation the *Refresh Bus Endpoint* will be called. It refreshes *default properties* 
+  to the database values in all UniConfig instances connected via kafka refresh topic.
+* At application runtime, if user uses *update-properties RPC* with *default properties* on input -> UniConfig will
+  update the properties in database and then use the *Refresh Bus Endpoint* call, which will reload properties in all
+  connected via kafka UniConfig instances.
+
+How to disable UniConfig Cloud Config and how the properties will be set up:
+
+![startup-without-ucc](startup-without-ucc.jpg)
+
+* First, before starting UniConfig, we must disable Cloud Config using the following properties
+
+```
+# Uniconfig cloud settings - depends on kafka connection
+uniconfig.cloud.config.enabled=false
+uniconfig.cloud.config.fail-fast=false
+uniconfig.cloud.config.application-name=uniconfig
+spring.cloud.bus.enabled=false
+spring.kafka.bootstrap-servers=http://localhost:9092
+management.endpoints.web.exposure.include=bus-refresh
+# Disable cloud connection to Kafka for testing purpose, property spring.cloud.bus.enabled must be false
+spring.autoconfigure.exclude=org.springframework.cloud.stream.function.FunctionConfiguration
+```
+
+* When UniConfig starts, it will first check if there are any *default properties* in the database that it could configure
+  * If *default properties* are in the database -> UniConfig instance will use it
+  * If not -> UniConfig instance will use its properties and, once loaded, will save them to a database
+  for a following UniConfig instances.
+* At the end of Spring initialisation the *Refresh Bus Endpoint* wouldn't be called.
```

Comment:
Then why is it in the diagram ?

Reply:
because if you look at it properly with UCC we have 2 property refreshes -> first one is manualy reading from db while startup and second one is by refresh bus endpoint functionality (but its the same result -> refresh properties)
and without UCC we just have 1 manualy reading from db while UC startup.

Reply:
rewritten